simple-generated/src/main.rs

```diff
@@ -1,45 +1,111 @@
-use std::io;
+use std::time::Duration;
 
-use ratatui::{backend::CrosstermBackend, Terminal};
-
-use crate::{
-    app::{App, AppResult},
-    event::{Event, EventHandler},
-    handler::handle_key_events,
-    tui::Tui,
+use color_eyre::Result;
+use crossterm::event::{KeyCode, KeyEvent, KeyModifiers};
+use event::{Event, EventSource};
+use ratatui::{
+    layout::Alignment,
+    style::{Style, Stylize},
+    widgets::{Block, BorderType, Paragraph},
+    DefaultTerminal, Frame,
 };
 
-pub mod app;
 pub mod event;
-pub mod handler;
-pub mod tui;
-pub mod ui;
-
-fn main() -> AppResult<()> {
-    // Create an application.
-    let mut app = App::new();
-
-    // Initialize the terminal user interface.
-    let backend = CrosstermBackend::new(io::stdout());
-    let terminal = Terminal::new(backend)?;
-    let events = EventHandler::new(250);
```

Comment:
I pushed the event creation into the app, as main doesn't really need to know about it. It does need to know about terminal init / restore generally though as it's easier to handle restore nicely by return a result from the app method.